### PR TITLE
Wen's API Service

### PR DESCRIPTION
**Pull request by wwenzz**
## Netlify link
https://wen-api-service.onrender.com/

## Collaborators
[wwenzz]

### CODE REVIEW

**Review by AntonellaMorittu — CHANGES_REQUESTED** [review]
Hello Wen, well done completing your first API! 👏 Your code meets almost all the requirements, I am curious how did you end up adding an `index.ejs` file to list your endpoints? Using `expressListEndpoints` will take care of it, see the [example](https://express-demo-bu5u.onrender.com/) provided for this week's assignment. You want to get a list of clickable endpoints with this readable format. Please fix that and you're good to go 🚀

**Review by AntonellaMorittu — APPROVED** [review]

File: server.js [hunk 1]
```diff
@@ -1,28 +1,133 @@
...
+// middleware: if homepage gets any query params -> throw 400 error
+app.get("/songs/:songId", queryParamContoller, (req, res, next) => {
+  const songId = req.params.songId;
+  const song = topMusicData.find(song => song.id === +songId);
+  if (song) {
+    res.json(song);
+  } else {
+    const err = new Error(
+      `There are no songs matching the id:${songId}... Make sure you are inquering id from 1 - 50.`
+    );
```
Comment (almaherris) [inline server.js:117]: Good that you're handling unavailable requests and I like that you are suggesting to the user how to do it correctly! Very user friendly 👍

File: server.js [hunk 2]
```diff
@@ -1,28 +1,133 @@
...
+// middleware: if homepage gets any query params -> throw 400 error
+app.get("/", queryParamContoller, (req, res) => {
+  let endpoints = expressListEndpoints(app);
+  endpoints.map(
+    (endpoint, index) => (endpoint.description = endpointDescr[index])
+  );
+  res.render("index.ejs", { data: endpoints });
+});
+
+// songs endpoint: display all songs and filter songs based on genre and/or artist
```
Comment (almaherris) [inline server.js:69]: All your endpoints seem to be working well, and the code looks clean!

**Review by almaherris — COMMENTED** [review]
Clean code with nice simple frontend to your API instructions. Good job as always Wen 🤩

**Pull request closed without merging**